src/basics.md

```diff
@@ -1244,6 +1245,64 @@ fn main() {
 }
 ```
 
+[ex-error-chain-simple-error-handling]: #ex-error-chain-simple-error-handling
+<a name="ex-error-chain-simple-error-handling"></a>
+## Handle errors correctly in main
+
+[![error-chain-badge]][error-chain] [![cat-rust-patterns-badge]][cat-rust-patterns]
+
+Handles error that occur when trying to open a file that does not
+exist. It is achieved by using [error-chain], a library that takes
+care of a lot of boilerplate code needed in order to [handle errors in
+Rust].
+
+`Io(std::io::Error)` inside [`foreign_links`] allows automatic
+conversion from [`std::io::Error`] into [`error_chain!`] defined type
+implementing the [`Error`] trait.
+
+The below recipe will tell how long the system has been running by
+opening the Unix file `/proc/uptime` and parse the content to get the
+first number. `read_uptime` returns a `Result`, which is the type that
+represents a possibility of failure. We use pattern matching in main
```

Comment:
Are you referring to "We use pattern matching [...]"?

"We use pattern matching in main to handle that result, which in our case either an error or the uptime in seconds." -> "Pattern matching is used in main to handle that result. Here it can be an error or the uptime in seconds."
Is it a better wording?

Reply:
Fair enough. I am not a native english speaker. Thanks for clarifying that.